nobl9/resource_direct.go

```diff
@@ -294,6 +294,62 @@ func (s appDynamicsDirectSpec) UnmarshalSpec(d *schema.ResourceData, spec v1alph
 	return
 }
 
+// Azure Monitor Direct
+// https://docs.nobl9.com/Sources/azure-monitor#azure-monitor-direct
+const azureMonitorDirectType = "azuremonitor"
+
+type azureMonitorDirectSpec struct{}
+
+func (s azureMonitorDirectSpec) GetSchema() map[string]*schema.Schema {
+	azureMonitorSchema := map[string]*schema.Schema{
+		"tenant_id": {
+			Type:        schema.TypeString,
+			Description: "[required] | Azure Tenant ID.",
+			Required:    true,
+		},
+		"client_id": {
+			Type:        schema.TypeString,
+			Description: "[required] | Azure Application (client) ID.",
+			Required:    true,
+		},
+		"client_secret": {
+			Type:        schema.TypeString,
+			Description: "[required] | Azure Application (client) Secret.",
+			Computed:    true,
+			Optional:    true,
+			Sensitive:   true,
+			ValidateDiagFunc: validation.ToDiagFunc(
+				validation.StringIsNotEmpty,
+			),
```

Comment:
Is it not required by other fields?

Reply:
I did it the same as the rest of directs is done. 

Because this validation is attached to a secret value (client_secret) and the secret itself is configured as:

```terraform
Optional:    true,
Computed:    true,
Sensitive:   true,
ValidateDiagFunc: validation.ToDiagFunc(
	validation.StringIsNotEmpty,
),
```

* when we specify a new direct (unique name), then TF during `apply` will throw an error, since it finds there no resource on n9 side
* when we apply and already have a direct of specific name, then TF will let us go, because the resource on n9 side already has client_secret

This validation func is not required for other fields, because they're explicitly `Required`